pulsar-broker/src/test/java/org/apache/pulsar/client/impl/NegativeAcksTest.java

```diff
@@ -146,6 +146,9 @@ public void testNegativeAcks(boolean batching, boolean usePartitions, Subscripti
             consumer.negativeAcknowledge(msg);
         }
 
+        assertTrue(consumer instanceof ConsumerBase<String>);
+        assertEquals(((ConsumerBase<String>) consumer).getUnAckedMessageTracker().size(), 0);
```

Comment:
Actually this test does not verify the change.

<img width="936" alt="image" src="https://github.com/apache/pulsar/assets/18204803/1561205a-80c0-4876-8d49-90e20c3adb38">

From the screenshot above, you can see `consumer.unAckedMessageTracker.size()` is also 0 without the change in this PR.

Reply:
`ComsumerImpl` with  non-batched message is sure to pass this test.
Problems only happen in two situations described in Motivation

Reply:
Could you add a test to simulate the situations described in the motivation? Otherwise, regression could be introduced if no tests can protect this change.

Reply:
It's already covered by `NegativeAcksTest#variations()`.
```java
                // batching / partitions / subscription-type / negativeAck-delay-ms / ack-timeout
                { false, false, SubscriptionType.Shared, 100, 1000 },
                { false, false, SubscriptionType.Failover, 100, 1000 },
                { false, true, SubscriptionType.Shared, 100, 1000 },
                { false, true, SubscriptionType.Failover, 100, 1000 },
                { true, false, SubscriptionType.Shared, 100, 1000 },
                { true, false, SubscriptionType.Failover, 100, 1000 },
                { true, true, SubscriptionType.Shared, 100, 1000 },
                { true, true, SubscriptionType.Failover, 100, 1000 },
```
1. For `MultiTopicsConsumerImpl`:  the second parameter is true means this is a  `MultiTopicsConsumerImpl`

3. For `ConsumerImpl` with the batched message:  the first parameter is true and the second is false means this a test for `ConsumerImpl` with the batched message